src/odemis/acq/align/fastem.py

```diff
@@ -279,76 +288,36 @@ def run_calibration(self, calibration):
-                                             self._detector, self._dataflow, self._ccd)
-            self.asm_config["descanner"]["scanOffset"] = d_offset
-
-        elif calibration == Calibrations.IMAGE_ROTATION_FINAL:
-            calibration.value.run(self._scanner, self._multibeam, self._descanner,
-                                  self._detector, self._dataflow, self._det_rotator)
-
-        elif calibration == Calibrations.IMAGE_TRANSLATION_FINAL:
-            d_offset = calibration.value.run(self._scanner, self._multibeam, self._descanner,
-                                             self._detector, self._dataflow, self._ccd)
-            self.asm_config["descanner"]["scanOffset"] = d_offset
-
-        elif calibration == Calibrations.DARK_OFFSET:
-            dark_offset = calibration.value.run(self._scanner, self._multibeam, self._descanner,
-                                                self._detector, self._dataflow)
-            self.asm_config["mppc"]["cellDarkOffset"] = dark_offset
-
-        elif calibration == Calibrations.DIGITAL_GAIN:
-            digital_gain = calibration.value.run(self._scanner, self._multibeam, self._descanner,
-                                                 self._detector, self._dataflow)
-            self.asm_config["mppc"]["cellDigitalGain"] = digital_gain
-
-        elif calibration == Calibrations.SCAN_ROTATION_FINAL:
-            calibration.value.run(self._scanner, self._multibeam, self._descanner, self._detector, self._dataflow)
-
-        elif calibration == Calibrations.SCAN_AMPLITUDE_FINAL:
-            s_offset, s_amplitude = calibration.value.run(self._scanner, self._multibeam, self._descanner,
-                                                          self._detector, self._dataflow)
-            self.asm_config["multibeam"]["scanOffset"] = s_offset
-            self.asm_config["multibeam"]["scanAmplitude"] = s_amplitude
-
-        elif calibration == Calibrations.CELL_TRANSLATION:
-            translation, resolution, cell_complete_resolution = calibration.value.run(self._scanner, self._multibeam,
-                                                                                      self._descanner, self._detector,
-                                                                                      self._dataflow,
-                                                                                      plot_cell_translation=True)
-            self.asm_config["mppc"]["cellTranslation"] = translation
-            self.asm_config["multibeam"]["resolution"] = resolution
-            self.asm_config["mppc"]["cellCompleteResolution"] = cell_complete_resolution
-
-        else:
-            raise ValueError(f"Unknown calibration {calibration.name}.")
+        calibration.run()
+
+        if isinstance(calibration, Calibrations.SCAN_AMPLITUDE_PREALIGN.value):
+            self.asm_config["multibeam"]["scanOffset"] = calibration.orig_config["multibeam"]["scanOffset"]
```

Comment:
This is fine... but I think we could consider doing something even more fancy.
Now that the run() is always the same, all that is different is the values to store. Could the calibration class provide this information too, so that we can have one generic code here?
Maybe something like `calibration.updated_settings` containing a list of tuples [str,str] (component role, VA name).
Then you'd do:
```
for comp, va in calibration.updated_settings:
      self.asm_config[comp][va] = calibration.orig_config[comp][va]
```